cdap-event-reader-spi/src/main/java/io/cdap/cdap/spi/events/EventResult.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.spi.events;
+
+import java.util.function.Consumer;
+
+public interface EventResult<T> extends AutoCloseable {
```

Comment:
Missing java doc. Also add some detail on how the consumer will be used.

Reply:
Since the implementation is defined by the extension, I don't want to be too specific on how the consumer should be used. However, I agree some guidance is necessary. What do you think of the updated JavaDoc?